compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -975,13 +960,16 @@ impl<'context> Elaborator<'context> {
         &mut self,
         self_type: &UnresolvedType,
         module: LocalModuleId,
-        impls: &[(Vec<Ident>, Span, UnresolvedFunctions)],
+        impls: &mut [(Vec<Ident>, Span, UnresolvedFunctions)],
     ) {
         self.local_module = module;
 
         for (generics, span, unresolved) in impls {
             self.file = unresolved.file_id;
-            self.declare_method_on_struct(self_type, generics, false, unresolved, *span);
+            let old_generic_count = self.generics.len();
+            self.add_generics(generics);
+            self.declare_methods_on_struct(self_type, false, unresolved, *span);
+            self.generics.truncate(old_generic_count);
```

Comment:
```suggestion           
            self.recover_generics(|this| {
                 this.add_generics(generics);
                 this.declare_methods_on_struct(self_type, false, unresolved, *span);
            });
```
Any reason we didn't use `recover_generics` here?